tests/testthat/test-data-table.R

```diff
@@ -1,15 +1,50 @@
+skip_if_not_installed("data.table")
```

Comment:
No need for this; we assume that suggested packges are installed.

Reply:
data.table is currently under suggests. Would it be better to add it?

Reply:
I think you mean "data.table is currently NOT under suggests".

And, no, please do not add it! We have an ongoing battle in usethis: just because we provide some niceties for a package we do NOT take it on as a dependency. We've got various measures in place re: this re: many packages usethis has convenience functions for.

Reply:
So, to be clear, I think this line should not be added, but it has nothing to do with suggests.

Reply:
This is what mocking `check_installed()` is about in tests like these.

Reply:
Oops, yes, that's what I meant! I had some issues mocking this one, but I can't exactly remember what at the moment. I'll revisit.

Reply:
Oh, I included in the PR description. Nice, Past Malcolm!

> Because writing NAMESPACE requires loading which requires data.table, I also decided just to skip the tests if data.table is not installed. I tried to hack away at it via mocking, but it didn't seem worth it (or particularly robust in this case). I'm happy to hear other suggestions, though.

Any suggestions?

Reply:
Oh, I could mock `roxygen_update_ns()` not to do anything. I think that should be fine for this test, right? Let me try it

Reply:
😕
So I think this is a concrete downside to roxygenising from usethis.
Kind of related to things we discussed in https://github.com/r-lib/usethis/pull/1377

But yeah, if we keep doing that, you are going to have to skip. I would just bring it inside the tests. And we're going to have to do that in all similar situations.

Reply:
This re-raises the issue of whether we even want functions like `use_tibble()` and `use_data_table()` at all.

https://github.com/r-lib/usethis/issues/1046

Reply:
Add what?

Reply:
I'm missing something here. data.table is listed in suggests, so why wouldn't it be available when you run the tests?

Reply:
Oh it's _not_ listed in suggests. I think the easiest fix is to just list it.

Reply:
data.table is not listed in suggests. I made a typo earlier. HOWEVER, I have been able to mock this successfully, so I think this is resolved